### DISCUSSION TITLE
Bad test results running hubs enterprise, on c4.large 3x3 servers, only 4 users on a room, no pause database.

### DISCUSSION THREAD

**Discussion by vjFred** [mozilla/hubs#2880] [General Questions & Discussion] [answered]
I just have upgraded to hubs-enterprise, looking to launch an event for 25 users, in tests i found a lot of bugs, the server just have kicked out users from my almost empty test rooms, this in all devices, also avatars do not see each other often, and the loading times for a room is more than in https://hubs.mozilla.com/ :(  What can be possible causes for this?

**Comment by vjFred** [comment, marked as answer]
Robin_k_wToday at 11:14 AM

I'd love to see some of your users's logs from your tests to pinpoint what's going on. Can you tell me more about what a user sees when they get kicked out of a room? I've never heard of this. 

Though! If you have multiple rooms open (on the same browser), the previous session will forcefully close unless you press "Cancel" to keep that session open.

Can you tell me more about what getting forcibly kicked out of a room looks like?
How large are your scenes and avatar files?
Do you have videos playing?
Do you have a custom client?
Any console logs? 
Can you try scaling down to 2x2 with c4.larges and see if you have the same results? For 25 CCU, that should be more than sufficient. We recommend scaling vertically before scaling horizontally with more servers

**Reply by vjFred** [reply]
Can you tell me more about what getting forcibly kicked out of a room looks like?

**this appear on 2 of 4 users,usual avatars

![image](https://user-images.githubusercontent.com/69004885/90565432-860df280-e16c-11ea-9014-834d1c8e0e0e.png)


If you have multiple rooms open (on the same browser) ** Not the case
Do you have videos playing?                                              **No
Do you have a custom client?                                             **No
Any console logs?                                                                **No

Can you try scaling down to 2x2 with c4.larges and see if you have the same results? For 25 CCU, that should be more than sufficient. We recommend scaling vertically before scaling horizontally with more servers  ** ok trying this

**Reply by robinkwilson** [reply]
Thanks for trying the 2x2 infrastructure.

**A few more questions for you**
Are you using an avatar pack? Usually invisible avatars indicate the specific avatar doesn't work right. 
Does the Developer tools > Network tab show any red errors?

**Reply by robinkwilson** [reply]
From vjFred discord:
My hubs-enterprise multiserver was tested on: 4 users on standard pack avatars, 10mb scene, no videos, no custom client, not the same browser ( no logs for now)

**Reply by vjFred** [reply]
Checking the avatar issue,  for some reason, any avatar pack i import, looks as aproved but in the general avatar list, all of them show me failed. 
here looks good
![image](https://user-images.githubusercontent.com/69004885/90793443-5d533d80-e2d1-11ea-814c-a22646163ff4.png)
but in the main avatar list
![image](https://user-images.githubusercontent.com/69004885/90793613-8a075500-e2d1-11ea-865c-8b27ae24c4a0.png)

show me this.. what can be the cause? this happens on all avatar pack i tried, so i really don´t know if the avatar is working or not

**Reply by bemyevent** [reply]
> From vjFred discord:
> My hubs-enterprise multiserver was tested on: 4 users on standard pack avatars, 10mb scene, no videos, no custom client, not the same browser ( no logs for now)

And with you 2x2 server , how many users per room have you reach ?

**Reply by vjFred** [reply]
@robin-k-wilson @bemyevent 
Update:
We just have the test in a real case scenario.
Single Room, 21 CCU. trying the 2x2 infrastructure, on C4 large ( as robin suggested ) on a friday night kids party.

This time was really different, the overall performance of audio chat for all users was better, 
( the 20MB, room was loaded with video, audio, 3d models, and a lot of kids inside doing all the stuff) 
i can say the results were good this time.
We start servers 2 hours in advance, on a new Stack on AWS, created a week ago, just for this event.

Things we notice as bugs, will be reported aside.

About the question "Does the Developer tools > Network tab show any red errors?"  No in this current tests

**Reply by emclaren** [reply]
Hi @vjFred  - are you continuing to have issues, or is this now resolved 😀
